openmoc/process.py

```diff
@@ -910,6 +910,43 @@ def get_mesh_cell_indices(self, point):
         else:
             return mesh_x, mesh_y, mesh_z
 
+    @classmethod
+    def from_lattice(cls, lattice, division=1):
+        """Create a mesh from an existing lattice
+
+        Parameters
+        ----------
+        lattice : openmoc.Lattice
+            Rectangular lattice used as a template for this mesh
+        division : int
+            Number of mesh cells per lattice cell.
+            If not specified, there will be 1 mesh cell per lattice cell.
+
+        Returns
+        -------
+        openmoc.process.Mesh
+            Mesh instance
+
+        """
+        cv.check_type("lattice", lattice, openmoc.Lattice)
+        
```

Comment:
Add a check and a warning if someone uses a non uniform Lattice since only C++ meshes support it

Reply:
I went ahead and made it an error.